File: src/tp03/ej09/Ellipse.java
package tp03.ej09;

public class Ellipse extends Figure {
    private Point center;
    private double axisA;
    private double axisB;

    public Ellipse (Point center, double axisA, double axisB) {
        this.center = center;
        this.axisA = axisA;
        this.axisB = axisB;
    }

    public double perimeter() {
        return 2 * Math.PI * Math.sqrt(((axisA * axisA) + (axisB * axisB)) / 2);
    }

    public double area() {
        return Math.PI * axisA * axisB;
    }

    protected Point getCenter() {
        return center;
    }

    protected double getAxisA() {
        return axisA;
    }

    protected double getAxisB() {
        return axisB;
    }

    @Override
    public String toString() {
        return "Ellipse: [Centro: " + center + ", DMayor: " + Math.max(axisA, axisB) + ", DMenor: " + Math.min(axisA, axisB) + "]";
    }
}

File: src/tp08/ej04/ellipse.rb
class Ellipse < Figure
  def initialize(center, a, b)
    @center = center
    @a = a
    @b = b
  end

  def area
    # Using the formula for the area of an ellipse: A = π * a * b
    Math::PI * @a * @b
  end

  def perimeter 
    # Using the formula for the circumference of an ellipse: C ≈ 2 * π * sqrt((a^2 + b^2) / 2)
    2 * Math::PI * Math.sqrt((@a**2 + @b**2) / 2.0)
  end

  def to_s
    "Ellipse [ #{@center}, #{@a}, #{@b} ]"
  end
end
